src/map/battle.cpp

```diff
@@ -1126,6 +1128,55 @@ bool battle_status_block_damage(struct block_list *src, struct block_list *targe
 			status_change_end(target, SC_KYRIE);
 	}
 
+	if (sc) {
+		int element = 0;
+		if (flag & BF_WEAPON) {
+			struct status_data *sstatus = status_get_status_data(src);
+			if(sstatus->rhw.ele == ELE_NEUTRAL && sstatus->lhw.ele > sstatus->rhw.ele)
+				element = battle_get_weapon_element(d, src, target, skill_id, skill_lv, EQI_HAND_L, false);
+			else
+				element = battle_get_weapon_element(d, src, target, skill_id, skill_lv, EQI_HAND_R, false);
+		} else if(flag & BF_MAGIC)
+			element = battle_get_magic_element(src, target, skill_id, skill_lv, d->miscflag);
+		else
+			element = battle_get_misc_element(src, target, skill_id, skill_lv, d->miscflag);
+		
+		//for(int i = SC_IMMUNE_PROPERTY_NOTHING; i <= SC_IMMUNE_PROPERTY_UNDEAD; i++) { suggestion for after all boss skill PRs are merged?
+		//	if (sc->getSCE(i) && element == i - 916)
+		//		damage = 0;
+		//}
+		if ((sc->getSCE(SC_IMMUNE_PROPERTY_NOTHING) && element == ELE_NEUTRAL)
+		|| (sc->getSCE(SC_IMMUNE_PROPERTY_WATER) && element == ELE_WATER)
+		|| (sc->getSCE(SC_IMMUNE_PROPERTY_GROUND) && element == ELE_EARTH)
+		|| (sc->getSCE(SC_IMMUNE_PROPERTY_FIRE) && element == ELE_FIRE)
+		|| (sc->getSCE(SC_IMMUNE_PROPERTY_WIND) && element == ELE_WIND)
+		|| (sc->getSCE(SC_IMMUNE_PROPERTY_DARKNESS) && element == ELE_DARK)
+		|| (sc->getSCE(SC_IMMUNE_PROPERTY_SAINT) && element == ELE_HOLY)
+		|| (sc->getSCE(SC_IMMUNE_PROPERTY_POISON) && element == ELE_POISON)
+		|| (sc->getSCE(SC_IMMUNE_PROPERTY_TELEKINESIS) && element == ELE_GHOST)
+		|| (sc->getSCE(SC_IMMUNE_PROPERTY_UNDEAD) && element == ELE_UNDEAD))
+			damage = 0;
+	
+		if (sc->getSCE(SC_DAMAGE_HEAL)) {
```

Comment:
Yeah I think your way makes more sense.
I'll move to the bottom.

Reply:
For my sense, this skill intended to make low level boss harder, so it could be just heal without any block status checking.
We just should check how's the official behavior does.

Reply:
Yeah I'll move it to the bottom.
But this makes me keep thinking.
For me all make sense.